clang/test/CIR/CodeGen/attribute-annotate-multiple.cpp

```diff
@@ -0,0 +1,73 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=BEFORE
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-after=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=AFTER
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM
+double *a __attribute__((annotate("withargs", "21", 12 )));
+int *b __attribute__((annotate("withargs", "21", 12 )));
+void *c __attribute__((annotate("noargvar")));
+void foo(int i) __attribute__((annotate("noargfunc"))) {
+}
+// redeclare with more annotate
+void foo(int i) __attribute__((annotate("withargfunc", "os", 23 )));
+void bar() __attribute__((annotate("withargfunc", "os", 22))) {
+}
+
+// BEFORE: module @{{.*}}attribute-annotate-multiple.cpp" attributes {cir.lang =
+
+// BEFORE: cir.global  external @a = #cir.ptr<null> : !cir.ptr<!cir.double>
+// BEFORE-SAME: [#cir.annotation<name = "withargs", args = ["21", 12 : i32]>]
+// BEFORE: cir.global  external @b = #cir.ptr<null> : !cir.ptr<!s32i>
+// BEFORE-SAME: [#cir.annotation<name = "withargs", args = ["21", 12 : i32]>]
+// BEFORE: cir.global  external @c = #cir.ptr<null> : !cir.ptr<!void>
+// BEFORE-SAME: [#cir.annotation<name = "noargvar", args = []>]
+
+// BEFORE: cir.func  @_Z3fooi(%arg0: !s32i) attributes {annotations =
```

Comment:
Please create an alias for printing `AnnotateAttr`, see `::getAlias()` methods in `CIRAttrs.cpp` for examples. We also don't use the generic `attributes { ... }` for printing/parsting. When you add the alias, this should look something like: `@_Z3fooi(%arg0: !s32i) #cir.annotation(#some_alias_name)`.

Reply:
tried to add getAlias, and it doesn't help with this particular issue.  FuncOp has its own custom printer, so I'm address this issue of attributes { ... } by changing the custom printer.

Reply:
As a general rule we should never print anything with the default “attributes {…}” syntax, we should change the printer/parser to do the proper thing. Changing the custom printer/parser is one way to work on this when the assembly format is not described in CIROps.td